cylc/flow/suite_files.py

```diff
@@ -787,4 +797,239 @@ def get_cylc_run_abs_path(path):
     """
     if os.path.isabs(path):
         return path
-    return get_suite_run_dir(path)
+    return get_workflow_run_dir(path)
+
+
+def _open_install_log(rund):
+    """Open Cylc log handlers for an install."""
+    time_str = get_current_time_string(
+        override_use_utc=True, use_basic_format=True,
+        display_sub_seconds=False
+    )
+    rund = Path(rund).expanduser()
+    log_path = Path(
+        rund,
+        'log',
+        'install',
+        f"{time_str}-install.log")
+    log_parent_dir = log_path.parent
+    log_parent_dir.mkdir(exist_ok=True, parents=True)
+    handler = logging.FileHandler(log_path)
+    handler.setFormatter(CylcLogFormatter())
+    INSTALL_LOG.addHandler(handler)
+
+
+def _close_install_log():
+    """Close Cylc log handlers for a flow run."""
+    for handler in INSTALL_LOG.handlers:
+        try:
+            handler.close()
+        except IOError:
+            pass
+
+
+def get_rsync_rund_cmd(src, dst, restart=False):
+    """Create and return the rsync command used for cylc install/re-install.
+        Args:
+            src (str): file path location of source directory
+            dst (str): file path location of destination directory
+            restart (bool): indicate restart (--delete option added)
+
+        Return: rsync_cmd: command used for rsync.
+
+    """
+
+    rsync_cmd = ["rsync"]
+    rsync_cmd.append("-av")
+    if restart:
+        rsync_cmd.append('--delete')
+    ignore_dirs = ['.git', '.svn', '.cylcignore']
+    for exclude in ignore_dirs:
+        if Path(src).joinpath(exclude).exists():
+            rsync_cmd.append(f"--exclude={exclude}")
+    if Path(src).joinpath('.cylcignore').exists():
+        rsync_cmd.append("--exclude-from=.cylcignore")
+    rsync_cmd.append(f"{src}/")
+    rsync_cmd.append(f"{dst}/")
+
+    return rsync_cmd
+
+
+def install_workflow(flow_name=None, source=None, run_name=None,
```

Comment:
Can break this function down a bit, would make it possible to unit test bits of of (e.g. the runN bit).